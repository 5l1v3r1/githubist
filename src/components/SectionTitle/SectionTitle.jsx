// @flow

import React, { type Node } from 'react';
import cx from 'classnames';

import styles from './SectionTitle.scss';

type Props = {
  children: Node,
  subject?: string,
  className?: string,
};

const SectionTitle = ({ children, subject, className }: Props) => {
  const classNames = cx(styles.sectionTitle, className);

  return (
    <div className={classNames}>
      {subject && <small className={styles.subject}>{subject}</small>}
      <h2 className={styles.title}>{children}</h2>
    </div>
  );
};

SectionTitle.defaultProps = {
  subject: undefined,
  className: '',
};

export default SectionTitle;
