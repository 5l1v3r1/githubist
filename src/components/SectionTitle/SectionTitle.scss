@import '../../theme/theme';

.sectionTitle {
  text-align: center;
}

.subject {
  color: $primary-color;
  font-size: map-get($font-sizes, x-small);
  font-weight: 600;
  letter-spacing: px-to-rem(2px);
  text-transform: uppercase;
}

.title {
  margin: 0;
  color: $black-color;
  font-size: map-get($heading-sizes, h2);
  font-weight: 600;
  line-height: $heading-line-height;
}
