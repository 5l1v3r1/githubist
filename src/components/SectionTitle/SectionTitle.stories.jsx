// @flow

import React from 'react';
import { storiesOf } from '@storybook/react';

import SectionTitle from './SectionTitle';

storiesOf('SectionTitle', module)
  .add('default', () => <SectionTitle subject="Standings">Top 5</SectionTitle>)
  .add('without subject', () => <SectionTitle>Top 5</SectionTitle>);
