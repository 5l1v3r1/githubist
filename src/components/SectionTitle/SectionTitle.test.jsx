// @flow

import React from 'react';
import { shallow } from 'enzyme';

import SectionTitle from './SectionTitle';

it('renders correctly', () => {
  const wrapper = shallow(<SectionTitle subject="Subject">Hello world</SectionTitle>);

  expect(wrapper).toMatchSnapshot();
});

it('renders correctly without subject', () => {
  const wrapper = shallow(<SectionTitle>Hello world</SectionTitle>);

  expect(wrapper).toMatchSnapshot();
});

it('passes custom classnames', () => {
  const wrapper = shallow(<SectionTitle className="testClassName">Hello world</SectionTitle>);

  expect(wrapper.find('div').hasClass('testClassName')).toBe(true);
});
