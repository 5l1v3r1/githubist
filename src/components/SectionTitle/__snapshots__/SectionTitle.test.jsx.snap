// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className=""
>
  <small>
    Subject
  </small>
  <h2>
    Hello world
  </h2>
</div>
`;

exports[`renders correctly without subject 1`] = `
<div
  className=""
>
  <h2>
    Hello world
  </h2>
</div>
`;
